mautic/mautic
# Fix Error on campaigns trigger for sm

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)         | Y  
| New feature/enhancement? (use the a.x branch)      | N
| Deprecations?                          | N
| BC breaks? (use the c.x branch)        | N
| Automated tests included?              | N
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #13646 



## Description

This fix ensures contacts are managed by EntityManager before save operation is issued so campaign sending of SMS does not fail anymore






---
### 📋 Steps to test this PR:


## To test executioner

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create triggered text message.
3. Create segment and add a contact with mobile number to it.
4. Create a campaign and select segment as source
5. Add Acton to send SMS
6. Publish the campaign
7. Run bin/consle mautic:campaigns:update && bin/consle mautic:campaigns:trigger
8. There should be no errors when you run the commands in previous step

## To test campaign event log

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create triggered text message.
3. Create segment and add a contact with mobile number to it.
4. Create campaign 1 with minimal configuration
5. Create a campaign 2 and select segment as source
6. Add Acton to send SMS
7. Add change campaign membership action and move contacts to campaign 1
8. Publish the campaigns
9. Run bin/consle mautic:campaigns:update && bin/consle mautic:campaigns:trigger
10. There should be no errors when you run the commands in previous step
